module/zfs/noload.c

```diff
@@ -143,7 +143,7 @@ static ssize_t __noload_run(struct nvme_algo *alg, abd_t *src, void *dst,
 	bio_map_buf(bio_dst, dst, d_len);
 
 	ret = nvme_algo_run(alg, bio_src, s_len, bio_dst, &out_len);
-	if (ret) {
+	if (ret || out_len > d_len) {
```

Comment:
Why should this be treated as a failure? I out_len > d_len the result is thrown out... Infact on failure we just return s_len anyway...

Reply:
I'm probably separating things that shouldn't be, it's a failure logically since we don't actually have that much data. But you're right in that it doesn't kill anything in the existing code.  It messed with my bounce copy logic in my original buffer patch but I will revisit that once I go over it again after Kelly is finished his initial tests.